### ISSUE TITLE
[core/win] ray.init is hanging forever on windows (vscode)

### ISSUE THREAD

**Issue by DanielAtKrypton** [ray-project/ray#12481]
### What is the problem?

`ray.init()` doesn't work. It starts and hangs forever as can be seen below:

![Imgur](https://i.imgur.com/jEOfz9z.png)

*Ray version and other system information (Python version, TensorFlow version, OS):*

ray version: `1.1.0.dev0` and `1.0.1.post1`

## Vs Code
![Imgur](https://i.imgur.com/Y7xggyM.png)

## Python dependencies:

[requiremets lock](https://github.com/krypton-unite/time_series_predictor/blob/ray_tune/requirements-lock.txt)

### Reproduction (REQUIRED)
Please provide a script that can be run to reproduce the issue. The script should have **no external library dependencies** (i.e., use fake or mock data / environments):

Originally discovered by using `tune_sklearn`, then smallest instructions to reproduce were found by @richardliaw . More information can be found [here](https://github.com/ray-project/tune-sklearn/issues/153).

#### Problem reproduction

```python
import ray
ray.init()
```

If we cannot run your script, we cannot fix your issue.

- [✔️] I have verified my script runs in a clean environment and reproduces the issue.
- [✔️] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by richardliaw** [comment]
One last thing, can you provide also the zip from `/tmp/ray/session_latest/logs`?

**Comment by DanielAtKrypton** [comment]
> One last thing, can you provide also the zip from `/tmp/ray/session_latest/logs`?

Sure. [Here](https://ufile.io/to4m9djy) you are.

**Comment by DanielAtKrypton** [comment]
Pay attention that it also hangs when run by the terminal.

**Comment by richardliaw** [comment]
@DanielAtKrypton hmm the link isn't working -- I think you can just upload it to the github issue?

**Comment by DanielAtKrypton** [comment]
> @DanielAtKrypton hmm the link isn't working -- I think you can just upload it to the github issue?

Sure.


[ray.zip](https://github.com/ray-project/ray/files/5685665/ray.zip)

**Comment by richardliaw** [comment]
OK, looks like we're seeing this:

```
I1128 01:19:05.817958 26028  9820 io_service_pool.cc:36] IOServicePool is running with 1 io_service.
I1128 01:19:05.818956 26028  9820 store_runner.cc:30] Allowing the Plasma store to use up to 4.85823GB of memory.
I1128 01:19:05.819965 26028  9820 store_runner.cc:44] Starting object store with directory C:\Users\Daniel\AppData\Local\Temp and huge page support disabled
I1128 01:19:05.856093 26028  9820 grpc_server.cc:74] ObjectManager server started, listening on port 58146.
I1128 01:19:05.915802 26028  9820 node_manager.cc:178] Initializing NodeManager with ID 029248c47777efa5284b913dc34551078a5ca865
I1128 01:19:05.917320 26028  9820 grpc_server.cc:74] NodeManager server started, listening on port 59204.
I1128 01:19:05.978418 26028  9820 agent_manager.cc:40] Not starting agent, the agent command is empty.
I1128 01:19:06.030061 26028  9820 service_based_accessor.cc:775] Received notification for node id = 029248c47777efa5284b913dc34551078a5ca865, IsAlive = 1
I1128 01:19:06.418423 26028  9820 worker.cc:123] Assigned worker 01000000ffffffffffffffffffffffffffffffff to job 01000000
W1128 01:19:06.678421 26028  9820 client_connection.cc:395] [worker]ProcessMessage with type RegisterClientRequest took 260 ms.
W1128 01:19:13.171999 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 22844
I1128 01:19:13.176996 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.221997 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 20156
W1128 01:19:13.222997 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 24024
I1128 01:19:13.227998 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
I1128 01:19:13.227998 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.394001 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 20584
I1128 01:19:13.398994 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.440003 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 19568
I1128 01:19:13.445996 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.502003 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 6812
I1128 01:19:13.506995 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.580003 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 24892
I1128 01:19:13.586000 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
W1128 01:19:13.722995 26028  9820 worker_pool.cc:454] Received a register request from an unknown worker 18424
I1128 01:19:13.728004 26028  9820 node_manager.cc:1380] Ignoring client disconnect because the client has already been disconnected.
I1128 01:19:36.431042 26028  9820 worker_pool.cc:359] Some workers of the worker process(25096) have not registered to raylet within timeout.
I1128 01:19:36.446074 26028  9820 worker_pool.cc:359] Some workers of the worker process(18760) have not registered to raylet within timeout.
I1128 01:19:36.490875 26028  9820 worker_pool.cc:359] Some workers of the worker process(19616) have not registered to raylet within timeout.
I1128 01:19:36.552772 26028  9820 worker_pool.cc:359] Some workers of the worker process(16484) have not registered to raylet within timeout.
I1128 01:19:36.567771 26028  9820 worker_pool.cc:359] Some workers of the worker process(22556) have not registered to raylet within timeout.
I1128 01:19:36.599239 26028  9820 worker_pool.cc:359] Some workers of the worker process(16024) have not registered to raylet within timeout.
I1128 01:19:36.646158 26028  9820 worker_pool.cc:359] Some workers of the worker process(26088) have not registered to raylet within timeout.
I1128 01:19:36.692728 26028  9820 worker_pool.cc:359] Some workers of the worker process(22000) have not registered to raylet within timeout.
```

@DanielAtKrypton do you happen to have a firewall that might be blocking ports? 
@rkooo567 do you know what's the setting to configure to get more internal ray logs?

**Comment by DanielAtKrypton** [comment]
> @DanielAtKrypton do you happen to have a firewall that might be blocking ports?
I am using a VPN tunnel and port forwarding.

**Comment by Dias2019** [comment]
so did you found the solution for the above problem?

**Comment by DanielAtKrypton** [comment]
> 
> 
> so did you found the solution for the above problem?

Not yet @Dias2019 . I decided to remove the feature in the meanwhile there isn't a solution.

**Comment by nirvana-msu** [comment]
I have the exact same problem as described above.. `ray.init()` hangs for me on Windows 7 while calling [ray._raylet.CoreWorker](https://github.com/ray-project/ray/blob/c5ae30d1d4167aa7ef0f7a08f83fee6cc085f036/python/ray/worker.py#L1171-L1176). This happens in all the following cases:
 * Started via Windows cmd
 * Started via Git Bash
 * Started via PyCharm in Run mode
 * Started via PyCharm in Debug mode

Console output:
```
D:\dev\ray\venv\Scripts\python.exe D:/dev/ray/main.py
2021-01-08 17:33:40,685	INFO services.py:1172 -- View the Ray dashboard at http://127.0.0.1:8265
```
Same happens when started without dashboard. I've turned off Antivirus, just in case.

Logs: [session_2021-01-11_17-27-51_675736_2308.zip](https://github.com/ray-project/ray/files/5797431/session_2021-01-11_17-27-51_675736_2308.zip)

**Comment by nirvana-msu** [comment]
Just tried installing the latest nightly build instead with `pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-1.2.0.dev0-cp38-cp38-win_amd64.whl`

Running the same script:
```python
import ray
ray.init(include_dashboard=False)
```

Now it doesn't even go that far. `raylet.out` shows the following stack trace:
```
[2021-01-12 14:26:10,055 I 13956 7216] io_service_pool.cc:36: IOServicePool is running with 1 io_service.
[2021-01-12 14:26:10,184 I 13956 7216] store_runner.cc:29: Allowing the Plasma store to use up to 0.559049GB of memory.
[2021-01-12 14:26:10,184 I 13956 7216] store_runner.cc:42: Starting object store with directory C:\Users\Nirvana\AppData\Local\Temp and huge page support disabled
[2021-01-12 14:26:10,187 I 13956 7216] grpc_server.cc:74: ObjectManager server started, listening on port 36650.
[2021-01-12 14:26:10,189 I 13956 7216] node_manager.cc:186: Initializing NodeManager with ID 8c084c901eb768adcfcc370ec8f492ff812ab328da2b1fd1c07d716d
[2021-01-12 14:26:10,190 I 13956 7216] grpc_server.cc:74: NodeManager server started, listening on port 55577.
[2021-01-12 14:26:10,202 I 13956 13664] agent_manager.cc:75: Monitor agent process with pid 14212, register timeout 30000ms.
[2021-01-12 14:26:10,212 I 13956 7216] raylet.cc:132: Raylet of id, 8c084c901eb768adcfcc370ec8f492ff812ab328da2b1fd1c07d716d started. Raylet consists of node_manager and object_manager. node_manager address: 192.168.1.184:55577 object_manager address: 192.168.1.184:36650 hostname: 192.168.1.184
[2021-01-12 14:26:10,215 I 13956 7216] service_based_accessor.cc:564: Received notification for node id = 8c084c901eb768adcfcc370ec8f492ff812ab328da2b1fd1c07d716d, IsAlive = 1
[2021-01-12 14:26:11,421 W 13956 7216] worker_pool.cc:409: Received a register request from an unknown worker 9304
[2021-01-12 14:26:11,421 C 13956 7216] node_manager.cc:1031:  Check failed: new_scheduler_enabled_ == false 
[2021-01-12 14:26:11,421 E 13956 7216] logging.cc:415: *** Aborted at 1610461571 (unix time) try "date -d @1610461571" if you are using GNU date ***
[2021-01-12 14:26:11,424 E 13956 7216] logging.cc:415:     @      0x7fed7218b68 raise
[2021-01-12 14:26:11,425 E 13956 7216] logging.cc:415:     @      0x7fed72198e1 abort
[2021-01-12 14:26:11,425 E 13956 7216] logging.cc:415:     @        0x13f4773f9 public: void __cdecl google::NullStreamFatal::`vbase destructor'(void) __ptr64
[2021-01-12 14:26:11,425 E 13956 7216] logging.cc:415:     @        0x13f475cc1 public: virtual __cdecl google::NullStreamFatal::~NullStreamFatal(void) __ptr64
[2021-01-12 14:26:11,425 E 13956 7216] logging.cc:415:     @        0x13f26f449 public: class google::NullStream & __ptr64 __cdecl google::NullStream::stream(void) __ptr64
[2021-01-12 14:26:11,425 E 13956 7216] logging.cc:415:     @        0x13f28aae7 public: class google::NullStream & __ptr64 __cdecl google::NullStream::stream(void) __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f2884ac public: class google::NullStream & __ptr64 __cdecl google::NullStream::stream(void) __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f229979 public: class google::NullStream & __ptr64 __cdecl google::NullStream::stream(void) __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f458ead public: class google::LogMessage::LogStream * __ptr64 __cdecl google::LogMessage::LogStream::self(void)const __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f4571f7 public: class google::LogMessage::LogStream * __ptr64 __cdecl google::LogMessage::LogStream::self(void)const __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f45bc7a public: class google::LogMessage::LogStream * __ptr64 __cdecl google::LogMessage::LogStream::self(void)const __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f720f24 bool __cdecl google::Demangle(char const * __ptr64,char * __ptr64,int)
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f72471f bool __cdecl google::Demangle(char const * __ptr64,char * __ptr64,int)
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f723f8b bool __cdecl google::Demangle(char const * __ptr64,char * __ptr64,int)
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f217f03 public: class google::NullStream & __ptr64 __cdecl google::NullStream::stream(void) __ptr64
[2021-01-12 14:26:11,426 E 13956 7216] logging.cc:415:     @        0x13f755830 bool __cdecl google::Demangle(char const * __ptr64,char * __ptr64,int)
[2021-01-12 14:26:11,427 E 13956 7216] logging.cc:415:     @         0x774159cd BaseThreadInitThunk
[2021-01-12 14:26:11,430 E 13956 7216] logging.cc:415:     @         0x7764a2e1 RtlUserThreadStart
```

Full logs: [session_2021-01-12_14-26-09_265884_11184.zip](https://github.com/ray-project/ray/files/5802742/session_2021-01-12_14-26-09_265884_11184.zip)

Anything I could do to help debug this?

**Comment by nirvana-msu** [comment]
I've raised a new issue with a clear description of the root cause - please see #13794

**Comment by DanielAtKrypton** [comment]
Neither here or at the duplicate issue https://github.com/ray-project/ray/issues/13794, a solution was developed as of now. In my understanding the ability to isolate concerns and environment is paramount. Maybe `ray-project` folks do not think this way...

**Comment by suuyawu** [comment]
I met the same issue when I remotely use the HPC from my lab.

**Comment by HornetGuppy** [comment]
I am having same issue of hanging in (python 3.7, WinPy,Windows) in venv. Whereas when I tried downgrading to 1.0.0 it has a redis connection error (both with and without VPN).

**Comment by nima-siboni** [comment]
I have same problem

**Comment by ctrl-Felix** [comment]
In PyCharm it is working but not in vscode for me...
Would be great to get an update or even a fix!

**Comment by pcmoritz** [comment]
This issue should now be fully resolved in master and will be released in 1.10 (to be released in Jan '22). I'll close this issue for now but please let us know if the problem still persists with 1.10 once it is released :)

**Comment by ky1emoon** [comment]
Still hanging error occurs on [Ubuntu 20.04, Python 3.8, Ray 1.10]

It works on the base interpreter,

but using `venv` environment still occurs hanging error on` ray.init()`

Note that `virtualenv` environment dose not occur this hanging problem.

**Comment by czgdp1807** [comment]
I will look into this. A bit weird, but in conda environments as well, some scripts (which use ray APIs) hang (for example, `test_dataclient_disconnect`). The behaviour is random though. Following is the output,

```
[2022-02-24 10:57:33,252 E 1852 1760] core_worker.cc:120: Failed to register worker 01000000ffffffffffffffffffffffffffffffffffffffffffffffff to Raylet. IOError: [RayletClient] Unable to register worker with raylet. Unknown error
```

**Comment by LAITRUNGMINHDUC** [comment]
Hi, it's work for me with this Python code-line:
```
ray.init(ignore_reinit_error=True, include_dashboard=False)
```

**Comment by mattip** [comment]
The mention of virtualenv here makes me think this is related to #24968. Could any of the people reporting this problem try again with a [nightly](https://docs.ray.io/en/master/ray-overview/installation.html#daily-releases-nightlies) that has the fix mentioned?

**Comment by mattip** [comment]
I think that #24968 (which was reverted but should be merged soon as #25806) will solve the problem.

**Comment by mattip** [comment]
Could anyone reporting this problem try the [latest nightly](https://docs.ray.io/en/master/ray-overview/installation.html#daily-releases-nightlies) builds? #25806 was merged, which should have solved the problems using ray under a virtual environment.

**Comment by mattip** [comment]
I am closing this, since we did not get any response after #25806 was merged. Please open a __new issue__ if this returns, with a complete reproducer.
